sklearn/utils/tests/test_validation.py

```diff
@@ -464,6 +466,43 @@ def test_check_array_accept_sparse_no_exception():
     check_array(X_csr, accept_sparse=('csr',))
 
 
+@pytest.fixture(params=['csr', 'csc', 'coo', 'bsr'])
+def X_64bit(request):
+    X = sp.rand(20, 10, format=request.param)
+    for attr in ['indices', 'indptr', 'row', 'col']:
+        if hasattr(X, attr):
+            setattr(X, attr, getattr(X, attr).astype('int64'))
+    yield X
+
+
+def test_check_array_accept_large_sparse_no_exception(X_64bit):
+    # When large sparse are allowed
+    if LARGE_SPARSE_SUPPORTED:
```

Comment:
Maybe mark this with `pytest.mark.skipif(not LARGE_SPARSE_SUPPORTED)` instead? otherwise we will be generating 64 bit sparse in the fixture with a scipy that doesn't support it.

Reply:
No, we want to do this: we want to test what happens if the user passes one that they could not have constructed directly with scipy.